modApiServer/src/org/aion/api/server/rpc3/RPCMethods.java

```diff
@@ -259,13 +282,13 @@ private BlockDetails serializeBlockDetails(Block block) {
                                         info.getReceipt(), i, block.getHeader().getNumber()),
                                 transaction.getBeaconHash() == null? null : ByteArray.wrap(transaction.getBeaconHash())));
             }
-            return Collections.unmodifiableList(transactionDetails);
+            return transactionDetails.toArray(new TxDetails[0]);
```

Comment:
should return `transactionDetails.toArray(new TxDetails[transactionDetails.size()])` ?